controllers/drpolicy_controller.go

```diff
@@ -90,6 +97,67 @@ func (r *DRPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	return ctrl.Result{}, nil
 }
 
+func validate(ctx context.Context, drpolicy *ramen.DRPolicy, apiReader client.Reader,
+	client client.Client, objectStoreGetter ObjectStoreGetter, log logr.Logger,
+) error {
+	var (
+		conditionSetTrue  func(reason, message string) error
+		conditionSetFalse func(reason string, err error) error
+	)
+
+	if condition := util.DrpolicyValidatedConditionGet(drpolicy); condition != nil {
+		if condition.Status == metav1.ConditionTrue {
+			log.Info(`valid -> valid`)
+
+			return nil
+		}
+
+		conditionUpdate := func(status metav1.ConditionStatus, reason, message string) error {
+			util.ConditionUpdate(drpolicy, condition, status, reason, message)
+
+			return client.Status().Update(ctx, drpolicy)
+		}
+		conditionSetFalse = func(reason string, err error) error {
+			log.Info(`invalid -> invalid`)
```

Comment:
If a prior S3 profile was incorrect and was subsequently corrected, the error message won't change for the next incorrect profile?

NOTE: If we decide to update the condition, we should take care not to update the same messages and just change the timestamp, as this has caused frequent reconciles with VRG in the past.

Reply:
You're right.  It should change.   Also, good point about updating conditionally when the message changes.  Issue https://github.com/RamenDR/ramen/issues/318 tracks this.